tests/test_requests.py

```diff
@@ -451,6 +452,180 @@ async def handler(request):
     assert request.token is None
 
 
+def test_credentials(app, capfd):
```

Comment:
Can we break this up into multiple tests?

Reply:
done